compat/mingw.c

```diff
@@ -661,8 +661,18 @@ static int mingw_open_append(wchar_t const *wfilename, int oflags, ...)
 	handle = CreateFileW(wfilename, FILE_APPEND_DATA,
 			FILE_SHARE_WRITE | FILE_SHARE_READ,
 			NULL, create, FILE_ATTRIBUTE_NORMAL, NULL);
-	if (handle == INVALID_HANDLE_VALUE)
-		return errno = err_win_to_posix(GetLastError()), -1;
+      
+        if (handle == INVALID_HANDLE_VALUE) {
+        
+	        DWORD err = GetLastError();
+
+		// Some network storage solutions (e.g. Isilon) might return ERROR_INVALID_PARAMETER 
+		// instead of expected error ERROR_PATH_NOT_FOUND, which results in a unknow error.
+		// If so, the error is now forced to be an ERROR_FILE_NOT_FOUND error instead.
+                if (err == ERROR_INVALID_PARAMETER)
+                        err = ERROR_FILE_NOT_FOUND;
```

Comment:
I missed in my original comment that the error should be `ERROR_PATH_NOT_FOUND`... While it does not make a huge difference from the point of view of the user-visible error message, it may make it clearer for developers reading this hunk what is actually happening if we use the correct error value?

Reply:
I was indeed also in doubt before my commit if it would not be better to use ERROR_PATH_NOT_FOUND instead, to make it clear what is really the issue. I should have checked this with you. But I will now also make the required change.